src/integrations/prefect-redis/prefect_redis/lease_storage.py

```diff
@@ -40,6 +44,69 @@ def _expiration_key(self, lease_id: UUID) -> str:
         """Generate Redis key for lease expiration."""
         return f"{self.expiration_prefix}{lease_id}"
 
+    @staticmethod
+    def _holder_key(holder: dict[str, Any] | None) -> str | None:
+        """Create a canonical holder key of form 'type:uuid'."""
+        if not holder:
+            return None
+        h_type = holder.get("type")
+        h_id = holder.get("id")
+        if not h_type or not h_id:
+            return None
+        return f"{h_type}:{h_id}"
+
+    @staticmethod
+    def _limit_holders_key(limit_id: UUID) -> str:
+        return f"prefect:concurrency:limit:{limit_id}:holders"
+
+    @staticmethod
+    def _holder_to_lease_key(limit_id: UUID, holder_key: str) -> str:
+        return f"prefect:concurrency:holder:{limit_id}:{holder_key}"
+
+    async def _ensure_scripts(self) -> None:
+        if self._create_sha is None:
+            create_script = """
+            -- KEYS[1] = lease_key
+            -- KEYS[2] = expirations_key
+            -- KEYS[3..n] = pairs of (limit_holders_key, holder_to_lease_key)
+            -- ARGV[1] = lease_json
+            -- ARGV[2] = expiration_ts (number)
+            -- ARGV[3] = lease_id
+            -- ARGV[4] = holder_key (or empty string)
+            redis.call('SET', KEYS[1], ARGV[1])
+            redis.call('ZADD', KEYS[2], ARGV[2], ARGV[3])
+            if ARGV[4] ~= '' then
+              local i = 3
+              while i <= #KEYS do
+                redis.call('SADD', KEYS[i], ARGV[4])
+                redis.call('SET', KEYS[i+1], ARGV[3])
+                i = i + 2
+              end
+            end
+            return 1
+            """
+            self._create_sha = await self.redis_client.script_load(create_script)
```

Comment:
Can we use `register_script`? TIL, that makes it so we won't need to send the script over the wire each time, which boosts performance.